app/serializers/api/v2/csv/heading_serializer.rb

```diff
@@ -0,0 +1,19 @@
+module Api
+  module V2
+    module Csv
+      class HeadingSerializer
+        include CsvSerializer
+
+        columns :goods_nomenclature_item_id,
+                :goods_nomenclature_sid,
+                :declarable,
+                :description,
+                :description_plain,
+                :formatted_description,
+                :producline_suffix
+
+        column(:leaf) { |heading| heading.commodities.empty? }
```

Comment:
This executes the query and instantiates all of the commodities, sadly. If you do `heading.commodities_dataset.count` you push the check down to sequel without instantiate every commodity that belongs to the heading :)